### ISSUE TITLE
Add `ignore: ["keyframe-selectors"]` to `selector-disallowed-list`

### ISSUE THREAD

**Issue by RobHannay** [stylelint/stylelint#7162]
### What minimal example or steps are needed to reproduce the bug?

```css
@keyframes fade-in {
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-disallowed-list": [
      ["/^[a-z]+$/"]
    ]
  }
}
```

### How did you run Stylelint?

[Demo here](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBrGBPAZgJwIYFsYBnAAk11gFoBLAOxOAB16zsJ8HmTuSIAHXGGoAXdPBIAGANxcSAX2azhETi279BIsSQCMMlgtpyQAGhCZqAGxgA5AnEQwAHgT7WAdGCJFT4CLQsAcwQQJhZGEGwAV2siCPEwnhIIohhrMGVsSihqIlxLSwgAdxgoSktc4XiSAG1ZJLqQAHoAPRrcSgAvAF0AagASJoju+pIRg2ZjM0gA6kCAMQhsfFxhEIArIn9fWD4fRETkkCJRawraKoQjy1XiS5NZFNO0umFKGaDKE9xaKFxsKDVCI3YR3CKTEByIA)

### Which Stylelint-related dependencies are you using?

```json
    "stylelint": "15.10.3",
    "stylelint-config-idiomatic-order": "9.0.0",
    "stylelint-config-prettier-scss": "1.0.0",
    "stylelint-config-recommended": "13.0.0",
    "stylelint-config-standard-scss": "10.0.0",
    "stylelint-prettier": "4.0.2",
```

### What did you expect to happen?

Keyframes values not to be included in this rule as they are not "selectors".

### What actually happened?

See demo link above:

```
error Unexpected selector "from"
    [(selector-disallowed-list)](https://stylelint.io/user-guide/rules/selector-disallowed-list)
    [2:3-7]

error Unexpected selector "to"
    [(selector-disallowed-list)](https://stylelint.io/user-guide/rules/selector-disallowed-list)
    [6:3-5]
```

### Do you have a proposal to fix the bug?

Either ignore `@keyframes` idents, or expand the rule's `ignore` option to include a `keyframes` option.

**Comment by ybiquitous** [comment]
@RobHannay Thanks for the report with the reproducible demo!

Personally, it's better to provide a new option to ignore keyframes. Because according to the [spec](https://drafts.csswg.org/css-animations/#keyframes), `from` or `to` are also a kind of *selectors*.
<img width="506" alt="image" src="https://github.com/stylelint/stylelint/assets/473530/96a846a8-6d87-4068-be10-474b30e97da6">

For examples:

Adding `{"ignore": "inside-keyframes"}`,

```json
{
  "selector-disallowed-list": [
    ["/^[a-z]+$/"],
    {"ignore": "inside-keyframes"}
  ]
}
```

Or, adding `ignoreAtRules`,

```json
{
  "selector-disallowed-list": [
    ["/^[a-z]+$/"],
    {"ignoreAtRules": ["keyframe"]}
  ]
}
```

I've marked this issue as *needs discussion*.

**Comment by ybiquitous** [comment]
FYI, the rule has already provided `ignore: ["inside-block"]` option:

https://github.com/stylelint/stylelint/blob/b509d18ff155d07a3c839639a15040654e47d9d7/lib/rules/selector-disallowed-list/README.md?plain=1#L100

**Comment by jeddy3** [comment]
A new option SGTM.

We've tended to reach for more specific options when distinguishing between subtypes of things (e.g. [different subtypes of _type selectors_](https://github.com/stylelint/stylelint/blob/dca4e789a8c15da76267d322fd4541513fcb6547/lib/rules/selector-max-type/README.md?plain=1#L82)). We can add an `ignore: ["keyframe-selectors"]` option to follow that convention.

**Comment by jeddy3** [comment]
I've labelled the issue as ready to implement. @RobHannay Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

**Comment by RobHannay** [comment]
Thank you everyone! 🙌
